### PR TITLE
Added Quick Sort and Interpolation Search

### PR DESCRIPTION
**Pull request by Kanishk-03-Jain**
I have added the python algorithm for Quick Sort.

### CODE REVIEW

**Review by FerdausPolok — APPROVED** [review]
LGTM!
Feel free to merge at your convenience. Thanks for your contribution here. 🎉

**Comment by Kanishk-03-Jain** [comment]
Hi @FerdausPolok ,

The pull request is approved and ready for merge. Since I don’t have write access, could you please merge it at your convenience?

Thank you!

File: algorithm/searching-algo-python/interpolation_search.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+print("Interpolation Search using Python")
+
+str = input("Please enter a list of sorted numbers (in ascending order) separated by spaces: ")
```
Comment (FerdausPolok) [inline algorithm/searching-algo-python/interpolation_search.py]:
```suggestion
inout_string= input("Please enter a list of sorted numbers (in ascending order) separated by spaces: ")
```
Let's avoid using the name str for a variable, as it shadows the built-in str function in Python.

**Review by FerdausPolok — CHANGES_REQUESTED** [review]
Also can we have below improvements?

**Edge Case for Single Element Lists:** The algorithm already handles this well with the check for list[hi] == list[lo]. However, it's a good idea to include a small condition to handle cases like an empty list or a single element list upfront.

**Positional Calculation:** Although this is rare, the current formula for pos can potentially result in a value outside the valid index range. Adding a safeguard to ensure pos remains within the list bounds can prevent possible errors.

**Review by FerdausPolok — COMMENTED** [review]
Also `quick_sort` just been added with someone else's PR. Could please resolve the conflicts? Thanks.

**Comment by Kanishk-03-Jain** [comment]
I have committed the improvements and also resolved the conflict, please check. @FerdausPolok 
Also please tell me any other changes or any inconsistency

Just to mention, I’m a beginner open-source contributor, and I would really appreciate your feedback to help me improve!

File: algorithm/sorting-algo-python/quick_sort.py [hunk 2]
```diff
@@ -8,7 +8,7 @@ def quick_sort(arr):
     return quick_sort(left) + middle + quick_sort(right)
 
 # Driver code to test above
-arr = [3, 6, 8, 10, 1, 2, 1]
+arr = [10, 2, 7, 3, 5, 8, 1, 5, 74, 2, 2, 7, 2, 8, 52, 6, 74, 2]
 print("Original array:", arr)
 sorted_arr = quick_sort(arr)
-print("Sorted array:", sorted_arr)
+print("Sorted array:", sorted_arr)
```
Comment (FerdausPolok) [inline algorithm/sorting-algo-python/quick_sort.py]: Add EOL. Formatting with Prettier will do it automatically.

**Review by FerdausPolok — CHANGES_REQUESTED** [review]
Can you also add a screenshot in the story description by running the files that shows the input output please?

File: algorithm/sorting-algo-python/quick_sort.py (hunk 2 above)
Comment (Kanishk-03-Jain) [reply algorithm/sorting-algo-python/quick_sort.py]: Prettier was not available for python, so I used black

**Comment by Kanishk-03-Jain** [comment]
Screenshot 2024-10-05 160513
Screenshot 2024-10-05 160905

There are the screenshots of the outputs for interpolation search and Quick sort

**Review by FerdausPolok — APPROVED** [review]

**Pull request merged**